### DISCUSSION TITLE
MacOS `kitty zsh` being non-login shell but `kitty $(which zsh)` is login shell

### DISCUSSION THREAD

**Discussion by SPiCaRiA** [kovidgoyal/kitty#7359] [Q&A] [answered]
Hi, I read somewhere else before (can't remember which issue :x) that Kitty by default opens as login shell under macOS to align with its wierd behavior. However, I feel that I don't want Kitty instances being subprocess of a login shell to mimick the same thing, which doesn't make much sense to me.

Not wanting to change the behavior of parent process being login shell, I am trying to alias kitty instead of setting the `shell` option in `kitty.conf`. Here comes the confusion: ideally, I want to create an alias like `kitty $SHELL` so it works regardless of the shell, but this (as well as `kitty $(which zsh)`) still starts as a login shell. `kitty zsh` works as a non-login shell though.

Thank you for any information regarding my question/purpose!

**Comment by kovidgoyal** [comment, marked as answer]
kitty instances are not subprocesses of a login shell. When kitty is
running the users *default systemwide login shell* it will run it using the login
process, in login mode. This is harmless beyond creating noise in the
process tree. If you dont want this behavior for some reason, then set
your default shell to something else and set shell in kitty to the
actual shell you want to run and it wont run in login mode.

**Reply by SPiCaRiA** [reply]
Thank you for the information, I guess setting `shell` to run in non-login mode is slightly different from what I want :(

Let me clarify why I want to do this:
When typing `kitty` in a Kitty instance, it inherits all the environment variables from the parent process, but still starting as a login shell. The problem of being a login shell here is that, on macOS, login shells will run `path_helper`, a binary intended to **initialize** paths. But when the login shell already has a `$PATH` inherited from the parent process (which generally shouldn't be the case?), `path_helper` messes up the order of the `$PATH`.

So ideally, I want Kitty instances **still start as login shells**, except **those called from another login shell** command line. It seems that setting alias like `kitty zsh` to avoid matching the full path of login shell (e.g. `/bin/zsh`) is the only way to do this?

**Comment by kovidgoyal** [comment]
The typical solution for this is to configure your path as you want it,
either in kitty.conf or in your shell rc files rather than depending on
path_helper.
For instance in my zshenv I have

```
# Add to PATH uniquely
typeset -U path
path=(~/.local/bin "$path[@]")
path=(~/bin "$path[@]")
```

Or define a functiona named kitty in your zshrc that clears PATH when
running kitty.

**Reply by SPiCaRiA** [reply]
Thank you for the reply! Yeah I'm doing something similar to this 🤣 (I'll mark your previous explanation as answer to align with my initial question about how kitty determines login shell)
